[stable-2.8] Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI (#59557)

sivel opened the pull request:
* Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI. Fixes #59548

* Restore cli additions
(cherry picked from commit 923e218)


Co-authored-by: Matt Martz <matt@sivel.net>

sivel committed:
[stable-2.8] Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI (#59557)

* Move plugin loader playbook dir additions back to Playbook instead of PlaybookCLI. Fixes #59548

* Restore cli additions
(cherry picked from commit 923e218)

Co-authored-by: Matt Martz <matt@sivel.net>

abadger commented on changelogs/fragments/59548-import-playbook-plugin-dirs.yml:
```diff
@@ -0,0 +1,4 @@
+bugfixes:
+- plugin loader - Move plugin loader playbook dir additions back to ``Playbook`` instead of ``PlaybookCLI``
```
From the terminology here (Move), it sounds like the code was removed from PlaybookCLI and added to Playbook.  I see that that was what was originally proposed but that it seems to have intentionally changed to be "in addition to" instead of moved.  If it's okay with you, I'll change this changelog entry to reflect that.

Also, I couldn't find the reason that the solution was changed to be used in both cases.  Just for my own curiousity, could you summarize that?

abadger replied:
https://github.com/ansible/ansible/pull/60330

sivel replied:
Yes, originally I was moving it, and didn't update the fragment later when I determined that it needed to exist in 2 places.

Changes were made in the become plugins PR to how/when plugins were loaded.  When I tested with a playbook relative inventory plugin (which is newly supported), loading it in the initial playbook via `Playbook.__init__` was too late, so it needed to be done in both the CLI and in the playbook.  This has the side effect of not being able to support inventory plugins from included playbook dirs if they are different than the parent, but that isn't a regression.

abadger committed:
Fix changelog entry for playbook_dir fix

The final version of the fix makes the addition in two places instead of
moving the single addition from one place to another

(cherry picked from commit 84e3c0ef1dac3ed52ec53dce5f77b3b5580355f5)

abadger commented:
Merged for the 2.8.4 release.
